src/interval.rs

```diff
@@ -55,16 +61,27 @@ pub struct Interval {
     //
     // Representations of zeros and NaNs are arbitrary; a zero can be either +0.0 or -0.0,
     // and a NaN can be either a qNaN or a sNaN with an arbitrary payload.
-    //
-    // In Debug formatting, the value of `rep` is printed as either
-    // `__m128d(-a, b)` (on x86-64) or `float64x2_t(-a, b)` (on AArch64).
     pub(crate) rep: F64X2,
 }
 
 unsafe impl Send for Interval {}
 unsafe impl Sync for Interval {}
 impl Unpin for Interval {}
 
+impl fmt::Debug for Interval {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        if self.is_empty() {
+            write!(f, "Interval::EMPTY")
+        } else {
+            let [a, b] = extract(self.rep);
+            f.debug_struct("Interval")
+                .field("inf", &-a)
+                .field("sup", &b)
+                .finish()
```

Comment:
How about using `inf()`/`sup()` so that [0, 0] is formatted as `Interval { inf: -0.0, sup: 0.0 }`?

Reply:
We could but IMHO for *debugging output* it could be misleading.  I think we really want to know the true content of the two bounds (if only to check that +0 and -0 are well handled by the operations).

Reply:
I think a reasonable choice would be either:

1. **Completely expose the internal representation**
   In this case, the empty interval should be formatted as `Interval { inf: NaN, sup: NaN }`.
1. **Hide the internal representation**
   In this case, [0, 0] should be formatted as `Interval { inf: -0.0, sup: 0.0 }`.

Reply:
For debugging purposes, I prefer the first choice.  I have pushed changes to the PR in that direction.